### PR TITLE
Remove duplicated http filters

### PR DESCRIPTION
**Pull request by zhaohuabing**
This PR removes duplicated HTTP filters from the HCM HTTP filter chain. Instead of creating one HTTP filter per route, now EG only creates one HTTP filter for each SecurityPolicy. The same filter will be enabled for all the applicable routes.

It also changes the filter suffix to the policy name to make it easier to debug.

Before:
name: envoy.filters.http.basic_auth/first-route

After:
name: envoy.filters.http.basic_auth/securitypolicy/default/policy-for-http-route-1

Fix https://github.com/envoyproxy/gateway/issues/2891

### CODE REVIEW

**Review by guydc — APPROVED** [review]

**Review by zirain — APPROVED** [review]

File: internal/gatewayapi/securitypolicy.go [hunk 1]
```diff
@@ -366,20 +366,24 @@ func (t *Translator) translateSecurityPolicyForRoute(
 	}
 
 	if policy.Spec.OIDC != nil {
-		if oidc, err = t.buildOIDC(policy, resources); err != nil {
+		if oidc, err = t.buildOIDC(
+			irConfigName(policy),
```
Comment (arkodg) [inline internal/gatewayapi/securitypolicy.go:370]: nit: curious why this passed for `buildOIDC` but computed for `buildBasicAuth`

Comment (zhaohuabing) [reply internal/gatewayapi/securitypolicy.go:370]: I missed that :-(  

It shouldn't be needed anymore.

https://github.com/envoyproxy/gateway/pull/2992 @arkodg

**Pull request merged**